src/java/org/apache/cassandra/io/util/File.java

```diff
@@ -123,9 +123,6 @@ public File(URI path)
      */
     public File(Path path)
     {
-        if (path != null && path.getFileSystem() != filesystem)
-            throw new IllegalArgumentException("Incompatible file system");
```

Comment:
I am wonder if the reason for this check  is an assumption that default file system stored in `filesystem` can be used in and is used in several API functions. If the file system is different from default, should the API functions not rely on the default? If I understood correctly, the default file system is used for `getPath`.

Reply:
This check prevents `File` being able to wrap a remote filesystem path; that is why the check is removed.
My checking of the uses of `filesystem` are:
* in constructors when a `Path` is not provided
* in `pathSeparator()`; this is potential issue, however all known filesystem implementations currently use the same separator value (`/`)
* in `toPath()` only when `path` is `null`

There is also some comment about this in the jira.